packages/node/src/evm/fulfillments/api-calls.ts

```diff
@@ -4,15 +4,32 @@ import { go } from '../../utils/promise-utils';
 import * as logger from '../../logger';
 import * as requests from '../../requests';
 import { DEFAULT_RETRY_TIMEOUT_MS } from '../../constants';
-import { ApiCall, Request, LogsErrorData, RequestErrorCode, RequestStatus, TransactionOptions } from '../../types';
+import { ApiCall, Request, LogsErrorData, RequestErrorMessage, RequestStatus, TransactionOptions } from '../../types';
 import { AirnodeRrp } from '../contracts';
+import { decodeRevertString } from '..';
 
 const GAS_LIMIT = 500_000;
 
-type StaticResponse = { readonly callSuccess: boolean } | null;
+type StaticResponse = { readonly callSuccess: boolean; readonly callData: string } | null;
 
 type SubmitResponse = ethers.Transaction | null;
 
+type ApiCallWithSignature = ApiCall & { readonly signature: string };
+
+async function signResponseMessage(request: Request<ApiCall>, masterHDNode: ethers.utils.HDNode) {
+  const airnodeWallet = ethers.Wallet.fromMnemonic(masterHDNode.mnemonic!.phrase);
+  return await airnodeWallet.signMessage(
+    ethers.utils.arrayify(
+      ethers.utils.keccak256(
+        ethers.utils.solidityPack(
+          ['bytes32', 'bytes'],
+          [request.id, request.responseValue || ethers.constants.HashZero]
```

Comment:
Right, if we reach this far then it means that `responseValue` is not `undefined`, `null` or empty. I guess this is here just because `responseValue` is set as optional in `ApiCall` [interface](https://github.com/api3dao/airnode/blob/e2e6009c8a257566a05119b34eaa2bf38ef5b9ac/packages/node/src/types.ts#L92) and typescript will complain if we call `fulfill` with a possibly `undefined` arg.

It would be cool if we could enforce some sort of XOR between `responseValue` and `errorMessage` since one of those properties must always have a value.

Reply:
That is not easy to type. You essentially need to have two versions of APIResponse:
1) ApiSuccess
2) ApiFail

both extend the ApiBase and and in code you differentiate between the two.. Not sure, but it's probably out of scope to type it like this (and also not worth it)...

That said, if the value is defined for sure, just use `!` as that conveys this meaning...

Reply:
But only the api call lambda, which is probably fine. I think if we fulfilled the transaction with invalid data the consquences could be bigger.